modules/builtin/src/backend/variables/number.ts

```diff
@@ -11,35 +11,43 @@ class BoxedNumber implements BoxedVariable<number> {
   private _value?: number
   private _nbTurns?: number
 
-  constructor({ nbOfTurns, value }: BoxedVarContructor<number | string>) {
-    if (typeof value === 'boolean') {
-      this._value = value
-      this._confidence = 1
-    } else if (typeof value === 'string') {
-      this._value = Number(value)
-      this._confidence = 0.5
+  constructor({ nbOfTurns, value, confidence }: BoxedVarContructor<number>) {
+    if (value) {
+      this._nbTurns = nbOfTurns
+      this.trySet(value, confidence)
     }
-
-    this._nbTurns = nbOfTurns
   }
 
-  get value() {
-    return this._value
+  get value(): number {
+    return this._nbTurns !== 0 ? this._value : undefined
   }
 
-  set value(val) {
-    this._value = val
+  set value(newValue: number) {
+    this.trySet(newValue, 1)
   }
 
-  trySet(val: number, confidence: number) {
-    this._value = val
-    this._confidence = confidence
+  trySet(value: number, confidence: number) {
+    try {
+      if (typeof value === 'number') {
+        this._value = value
+        this._confidence = confidence
+      } else {
+        this._value = Number(value)
+        this._confidence = 0.5
```

Comment:
Catch not necessary here also. You could check if the result of the cast is NaN, which would mean the value was invalid, and the confidence could be set to 0 then. 

What do you think about adding the ability to extract numbers from a string ? For example, if you type `500$`, we could extract the numbers and set the confidence to something like 0.75 ?